### ISSUE TITLE
Incorrect Needs in 3.9.4

### ISSUE THREAD

**Issue by Zimgineering** [rwmt/Multiplayer#403]
After 3.9.4 a pawns needs are not entirely correct in multiplayer.

Converting to Singleplayer yields the correct needs:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/b16624aa-eae6-4bc4-8dcb-5082aea8fe06)
Converting that back to Multiplayer yields the incorrect needs:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/eb14fa75-0ab7-42f2-8251-b73249ecb968)

Reverting Multiplayer to 3.9.3 (3ec9c4c) resolves the issue.
---
Modlist (alphabetized)
- '[FSF] Complex Jobs'
- '[FSF] Filth Vanishes With Rain And Time'
- '[RF] Rainbeau Flambe - Storyteller'
- '[SR]Factional War'
- Alpha Animals
- Alpha Genes
- Alpha Memes
- Alpha Mythology
- Animal Controls
- Animals are fun
- Animals Logic
- Auto-Cut Blight
- BetterInfoCard
- Bionic icons
- Biotech
- Common Sense
- Conduit Deconstruct
- Core
- Cyber Fauna 1.4
- Damage Indicators [1.4]
- Dinosauria
- Dreamer's Dreams
- Dual Wield
- Dubs Mint Menus
- Dubs Mint Minimap
- Expanded Prosthetics and Organ Engineering - Forked
- Follow Me
- FSharp.Core
- Giddy-up! Caravan
- Giddy-up! Core
- Giddy-up! Ride and Roll
- Go Explore!
- Gunplay
- Harmony
- Harvest Organs Post Mortem
- HugsLib
- Humanoid Alien Races
- Ideology
- Immortals
- Infused 2 Expansion
- Infusion 2
- Interaction Bubbles
- Let's Trade! [1.4+]
- Mass Graves
- Megafauna
- Mining Priority
- Misc. Robots
- Misc. Training
- Misc. TurretBase, Objects
- Miscellaneous 'CORE'
- Mod Manager
- ModDiff
- Moody
- More Sculpture
- Multiplayer
- Multiplayer Compatibility
- Outfitted
- Pick Up And Haul
- ProxyHeat
- Quality Cooldown
- Ranged DPS (Continued)
- Recipe icons (Continued)
- Replace Stuff
- Research Reinvented
- Reunion
- RimFridge Updated
- RimHUD
- Rimsenal - Augmented Vanilla Pack
- Royalty
- RPG Style Inventory Revamped
- Run and Hide (Continued)
- RunAndGun
- Share The Load
- Simple sidearms
- Smart Medicine
- Smarter Construction
- Sparkling Worlds - Full Mod
- SRTS Expanded
- Stonecutting Extended
- Tech Advancing
- Utility Columns
- Vanilla Apparel Expanded
- Vanilla Apparel Expanded — Accessories
- Vanilla Armour Expanded
- Vanilla Books Expanded
- Vanilla Events Expanded
- Vanilla Expanded - Ideology Patches
- Vanilla Expanded - Royalty Patches
- Vanilla Expanded Framework
- Vanilla Factions Expanded - Ancients
- Vanilla Factions Expanded - Deserters
- Vanilla Factions Expanded - Empire
- Vanilla Factions Expanded - Pirates
- Vanilla Fishing Expanded
- Vanilla Furniture Expanded
- Vanilla Furniture Expanded - Farming
- Vanilla Furniture Expanded - Medical Module
- Vanilla Furniture Expanded - Power
- Vanilla Furniture Expanded - Production
- Vanilla Furniture Expanded - Security
- Vanilla Genetics Expanded
- Vanilla Ideology Expanded - Memes and Structures
- Vanilla Ideology Expanded - Relics and Artifacts
- Vanilla Outposts Expanded
- Vanilla Plants Expanded
- Vanilla Psycasts Expanded
- Vanilla Storytellers Expanded - Perry Persistent
- Vanilla Textures Expanded
- Vanilla Weapons Expanded
- Vanilla Weapons Expanded - Coilguns
- Vanilla Weapons Expanded - Grenades
- Vanilla Weapons Expanded - Heavy Weapons
- Vanilla Weapons Expanded - Laser
- Vanilla Weapons Expanded - Non-Lethal
- Wall Light
- Water Freezes
- Where is my weapon?
- Yet Another Stockpile Presets

**Comment by Zimgineering** [comment]
Issue persists, is there any additional info needed?

**Comment by notfood** [comment]
First and foremost, you need to test it with vanilla multiplayer only otherwise your bug report is invalid here.

**Comment by Zimgineering** [comment]
With only the Multiplayer, Multiplayer Compat, and Hugslib mods loaded we have roughly the same output:

Pre-convert to singleplayer:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/474526be-2090-469d-a063-26075d2f46c1)
Post-convert to singelplayer:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/a08ae9d5-4f35-4e08-a44d-bfd1aea973a4)

**Comment by notfood** [comment]
Can you try on a new save? I suspect your global time is off in that save, giving the needs negative time and making them disappear. It also manifests as psycasts having insane or no cooldowns. It's fixed by editing the save manually and searching for ticksGame and altering it until it's right.

**Comment by Zimgineering** [comment]
Fresh vanilla+MP save has the same issue:

Singleplayer:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/ebd5a506-61dc-4cbe-8836-a86a24b0d0bd)
Multiplayer:
![image](https://github.com/rwmt/Multiplayer/assets/49336941/ce30b03d-dbc9-497f-b5ed-96f85390ed95)

Used a fresh install as well to be certain there wasn't any lingering local mods or DLC corruption.

**Comment by Zetrith** [comment]
This is caused by changes in 3cf6b22 which were made to reduce desyncs. The difference should be purely visual and the list should become correct if you let the game tick for a while (can you confirm it?). We'll work on a proper fix

**Comment by Zimgineering** [comment]
Yup, confirmed that the needs are corrected after letting it tick for a bit (works in the modded save too) 

Thanks for the clarity!
